src/pz390.java

```diff
@@ -9436,23 +9488,32 @@ private void ins_E3XX(){   // RPI 1207
 		case 0x83: //  E383 RXYa MSGC R1,D2(X2,B2) RPI 2202 
 			psw_check = false;
 			ins_setup_rxy();
-			rlv1 = reg.getLong(rf1) * mem.getLong(xbd2_loc);
-			reg.putLong(rf1, rlv1);
-			if (rlv1 != reg.getLong(rf1) * mem.getLong(xbd2_loc) ) {
-				psw_cc = psw_cc3;
+			big_int1 = new BigInteger(get_signed_bytes(reg_byte, rf1, 8)); 
+																			
+			big_int2 = new BigInteger(get_signed_bytes(mem_byte, xbd2_loc, 8));
+			big_int1 = big_int1
+			         .multiply(big_int2);
+			rlv1 = big_int1.longValue();
+			fp_bi_to_wreg(reg_byte,rf1,big_int1, 8);
+			if (big_int1.compareTo(bi_max_pos_long) == 1
+			|| big_int1.compareTo(bi_min_neg_long) == -1) {                  
+                psw_cc = psw_cc3; 
+			} else if (rlv1 > 0){
+				psw_cc = psw_cc2;
+			} else if (rlv1 == 0) {
+				psw_cc = psw_cc0;
 			} else {
-				psw_cc = get_long_comp_cc(rlv1, 0);
-			}
-			break;
+				psw_cc = psw_cc1;
+			}               
+ 			break;
```

Comment:
See MSGRKC comment about setting the condition code and possible fixed-point overflow.